crates/bevy_ecs/src/system/query.rs

```diff
@@ -1248,7 +1248,8 @@ impl<'w, 's, D: QueryData, F: QueryFilter> Query<'w, 's, D, F> {
     ///
     /// ## Panics
     ///
-    /// This will panic if `NewD` is not a subset of the original fetch `Q`
+    /// If this `Query`'s fetch is not a subset of `NewD`,
+    /// for example, it will panic if `Q` may contain entities not contained in `NewD`.
```

Comment:
What is `Q` here? The generic variable for the Query's `QueryData` part changed to `D`, I suppose that is meant.

I agree that is not much more clear as well. "this `Query`" gives an idea what this is about, but it also increased the number of "things" to bring into relations from 2 to 3 in a single sentence.

My current understating from your comment on my linked issue is that `NewD` must no reduce the number of matched entities by additional filters or extra components, and must not have a "higher level" of access, like going from `&T` to `&mut T`.